packages/ui/src/json.ts

```diff
@@ -0,0 +1,186 @@
+/* @ts-nocheck */
+import components from '../src/importer'
+import apiOptions from './api-options'
+import { PropOptions } from 'vue-class-component'
+import en from '../../docs/src/locales/en/en.json'
+
+const path = require('path')
+const pack = require(path.join(__dirname, '../../package.json'))
+const fs = require('fs')
+
+type Prop = {
+  name: string,
+  type?: string | string[],
+  title: string
+  description?: string
+} & PropOptions
+
+type ComponentOption = {
+  name: string,
+  title: string
+  description?: string
+  type?: string | string[],
+}
+
+type Component = {
+  slots?: any[],
+  events?: any[],
+  methods?: any[],
+  props?: any[],
+  name: string
+  title: string
+  importIdentifier: string
+  description?: string
+}
+
+function splitCamelCase (input: string) {
+  return input.replace(/([a-z](?=[A-Z]))/g, '$1 ').toLowerCase()
+}
+
+function snakeToCamel (str: string) {
+  return str.toLowerCase().replace(/([-_][a-z])/g, group =>
+    group
+      .toUpperCase()
+      .replace('-', '')
+      .replace('_', ''),
+  )
+}
+
+function parseSlots (component: string) {
+  const slots = en.api[component]?.slots || en.all.api[component]?.slots
+  if (!slots) {
+    return null
+  }
+
+  return Object.keys(slots).map(key => {
+    return {
+      name: key,
+      title: splitCamelCase(key),
+      description: slots[key],
+    }
+  })
+}
+
+function parseProps (component: string, input: any) {
+  // eslint-disable-next-line
+  const props: object | undefined = en.api[component]?.props || en.all.api[component]?.props || {}
+  const componentApi = apiOptions[component]?.props as any
+
+  if (!props) {
+    return null
+  }
+
+  return Object.keys(props).map((key: string) => {
+    const value: Prop = {

```

Comment:
Also maybe some better naming? `propDescription` maybe? And type `PropDescription` instead of `Prop`